cmd/kubeapps-apis/server/plugins.go

```diff
@@ -109,19 +128,43 @@ func registerPlugins(pluginPaths []string, grpcReg grpc.ServiceRegistrar, gwArgs
 }
 
 // registerGRPC finds and calls the required function for registering the plugin for the GRPC server.
-func registerGRPC(p *plugin.Plugin, pluginPath string, registrar grpc.ServiceRegistrar) error {
+func (s *pluginsServer) registerGRPC(p *plugin.Plugin, pluginDetail *plugins.Plugin, registrar grpc.ServiceRegistrar) error {
 	grpcRegFn, err := p.Lookup(grpcRegisterFunction)
 	if err != nil {
-		return fmt.Errorf("unable to lookup %q for %q: %w", grpcRegisterFunction, pluginPath, err)
+		return fmt.Errorf("unable to lookup %q for %v: %w", grpcRegisterFunction, pluginDetail, err)
 	}
-	type grpcRegisterFunctionType = func(grpc.ServiceRegistrar)
+	type grpcRegisterFunctionType = func(grpc.ServiceRegistrar) interface{}
 	grpcFn, ok := grpcRegFn.(grpcRegisterFunctionType)
 	if !ok {
-		var dummyFn grpcRegisterFunctionType = func(grpc.ServiceRegistrar) {}
-		return fmt.Errorf("unable to use %q in plugin %q due to mismatched signature.\nwant: %T\ngot: %T", grpcRegisterFunction, pluginPath, dummyFn, grpcRegFn)
+		var dummyFn grpcRegisterFunctionType = func(grpc.ServiceRegistrar) interface{} { return nil }
+		return fmt.Errorf("unable to use %q in plugin %v due to mismatched signature.\nwant: %T\ngot: %T", grpcRegisterFunction, pluginDetail, dummyFn, grpcRegFn)
 	}
-	grpcFn(registrar)
+	server := grpcFn(registrar)
 
+	return s.registerPluginsSatisfyingCoreAPIs(server, pluginDetail)
+}
+
+// registerPluginsImplementingCoreAPIs checks a plugin implementation to see
+// if it implements a core api (such as `packages.v1alpha1`) and if so,
+// keeps a (typed) reference to the implementation for use on aggregate APIs.
+func (s *pluginsServer) registerPluginsSatisfyingCoreAPIs(pluginSrv interface{}, pluginDetail *plugins.Plugin) error {
+	// The following heck if the service implements an interface is what
+	// grpc-go itself does, see:
+	// https://github.com/grpc/grpc-go/blob/v1.38.0/server.go#L621
+	serverType := reflect.TypeOf(pluginSrv)
+	corePackagesType := reflect.TypeOf((*packages.PackagesServiceServer)(nil)).Elem()
+
+	if serverType.Implements(corePackagesType) {
+		pkgsSrv, ok := pluginSrv.(packages.PackagesServiceServer)
+		if !ok {
+			return fmt.Errorf("Unable to convert plugin %v to core PackagesServicesServer although it implements the same.", pluginDetail)
```

Comment:
This error should not be possible, since we've already checked that the pluginSrv does indeed implement the core packages interface, but I can't ignore it :)